### PR TITLE
Project Files in GUI

### PR DESCRIPTION
**Pull request by rinkk**
This updates the schema file for ogs project files and adds functionality for reading/writing projects in gui. this is currently limited to reading/writing input files becauses other aspects of an ogs project are not supporting in gui. 
In addition, [here are the changes to Tests/Data](https://github.com/ufz/ogs-data/commit/96b789808a081fbb5709f986622d15cd6b022600).

**Missing:** Due to the changes required for gui input, there are adjustment for reading project files in the console app necessary.

This solves #1121

### CODE REVIEW

File: FileIO/OpenGeoSysProject.xsd [hunk 1]
```diff
@@ -2,57 +2,183 @@
...
+          <xs:complexType>
+            <xs:sequence>
+              <xs:element name="mesh" type="MeshFileType" minOccurs="0" maxOccurs="unbounded" />
+              <xs:element name="geometry" type="FileType" minOccurs="0" maxOccurs="unbounded" />
+              <xs:element name="stations" type="FileType" minOccurs="0" maxOccurs="unbounded" />
+            </xs:sequence>
+          </xs:complexType>
+        </xs:element>
+
+        <xs:element name="processes" minOccurs="0" maxOccurs="1">
```
Comment (chleh) [inline FileIO/OpenGeoSysProject.xsd]: min 1

File: FileIO/OpenGeoSysProject.xsd [hunk 2]
```diff
@@ -2,57 +2,183 @@
...
+      <xs:element name="file" type="xs:string" minOccurs="1" maxOccurs="1" />
+      <xs:element name="type" type="xs:string" minOccurs="1" maxOccurs="1" />
+    </xs:all>
+  </xs:complexType>
+
+  <xs:complexType name="ProcessType">
+    <xs:sequence>
+      <xs:element name="name" type="xs:string" minOccurs="1" maxOccurs="1" />
+      <xs:element name="type" type="xs:string" minOccurs="1" maxOccurs="1" />
+      <xs:element name="process_variable" type="xs:string" minOccurs="1" maxOccurs="1" />
```
Comment (chleh) [inline FileIO/OpenGeoSysProject.xsd]: `<process_variable>` is subject to change, at least it's process specific.

File: FileIO/OpenGeoSysProject.xsd [hunk 3]
```diff
@@ -2,57 +2,183 @@
...
-          <xs:element name="geoReference" type="xs:string" minOccurs="0" />
-          <xs:element name="texture" type="xs:string" minOccurs="0" />
-        </xs:sequence>
-      </xs:extension>
-    </xs:complexContent>
+
+  <!-- output-object with required type-reference -->
+  <xs:complexType name="OutputFileType">
+    <xs:all>
+      <xs:element name="file" type="xs:string" minOccurs="1" maxOccurs="1" />
```
Comment (chleh) [inline FileIO/OpenGeoSysProject.xsd]: renamed to `prefix`, I think.

File: FileIO/OpenGeoSysProject.xsd (hunk 1 above)
Comment (rinkk) [inline FileIO/OpenGeoSysProject.xsd]: no, every child of the root element (except input files) need to have minOccurs=0 so you can build a project from input files.
once you want to start simulation you will need 1 but that needs to be checked in config tree.

File: FileIO/OpenGeoSysProject.xsd [hunk 4]
```diff
@@ -2,57 +2,183 @@
...
+      <!-- undefined content -->
+      <xs:any processContents="lax" maxOccurs="unbounded"/>
+    </xs:sequence>
+  </xs:complexType>
+
+  <xs:complexType name="ParameterType">
+    <xs:sequence>
+      <xs:element name="name" type="xs:string" minOccurs="1" maxOccurs="1" />
+      <xs:element name="type" type="xs:string" minOccurs="1" maxOccurs="1" />
+      <xs:element name="value" type="xs:decimal" minOccurs="1" maxOccurs="1" />
```
Comment (endJunction) [inline FileIO/OpenGeoSysProject.xsd]: "value" is depending on "type".
If type is MeshProperty, then there is no "value" field.

File: FileIO/OpenGeoSysProject.xsd [hunk 5]
```diff
@@ -2,57 +2,183 @@
...
+    </xs:sequence>
+  </xs:complexType>
+
+  <xs:complexType name="SourceTermType">
+    <xs:sequence>
+      <xs:element name="geometrical_set" type="xs:string" minOccurs="1" maxOccurs="1" />
+      <xs:element name="geometry" type="xs:string" minOccurs="1" maxOccurs="1" />
+      <xs:element name="type" type="xs:string" minOccurs="1" maxOccurs="1" />
+      <xs:element name="value" type="xs:decimal" minOccurs="1" maxOccurs="1" />
```
Comment (endJunction) [inline FileIO/OpenGeoSysProject.xsd]: The source terms are not yet handled. But there shall be also an option to use a mesh's property as a source term.

**Comment by endJunction** [comment]
No activity for more than 30 days.

**Pull request closed without merging**